build.rs

```diff
@@ -119,44 +121,123 @@ fn main() {
 
     // C header search paths
     let mut header_search_paths: Vec<PathBuf> = Vec::new();
-    let target_os = env_var("CARGO_CFG_TARGET_OS");
 
     if cfg!(feature = "use_system_capstone") {
-        #[cfg(feature = "use_system_capstone")]
+        #[cfg(windows)] panic!("Feature 'use_system_capstone' is not supported on Windows");
+        #[cfg(all(not(windows), feature = "use_system_capstone"))]
         {
             link_type = find_system_capstone(&mut header_search_paths);
         }
     } else {
         if cfg!(feature = "build_capstone_cmake") {
-            #[cfg(feature = "build_capstone_cmake")] cmake();
+            #[cfg(windows)] panic!("Feature 'build_capstone_cmake' is not supported on Windows");
+            #[cfg(all(not(windows), feature = "build_capstone_cmake"))] cmake();
         } else {
-            // TODO: need to add this argument for windows 64-bit, msvc, dunno, read
-            // COMPILE_MSVC.txt file cygwin-mingw64
-
-            // In BSDs, `make` does not refer to GNU make
-            let make_cmd = if target_os.contains("bsd") || target_os == "dragonfly" {
-                "gmake"
-            } else {
-                "make"
-            };
-
-            let out_dir = env_var("OUT_DIR");
-            Command::new(make_cmd)
-                .current_dir(CAPSTONE_DIR)
-                .status()
-                .expect("Failed to build Capstone library");
-            let capstone = "libcapstone.a";
-            Command::new("cp")
-                .current_dir(CAPSTONE_DIR)
-                .arg(&capstone)
-                .arg(&out_dir)
-                .status()
-                .expect("Failed to copy capstone library to OUT_DIR");
-
-            println!("cargo:rustc-link-search=native={}", out_dir);
+            #[cfg(windows)]
+            {
+                let use_static_crt = {
+                    let target_features = env::var("CARGO_CFG_TARGET_FEATURE").unwrap_or(String::new());
+                    target_features.split(",").any(|f| f == "crt-static")
+                };
+
+                let files = [
```

Comment:
Can we automate this list generation? Even adding a bash script to [`scripts/`](scripts) would be preferable. When we upgrade the underlying Capstone C library, a script will make it easier (even if we need to modify the script).

Reply:
I didn't notice the script you used to generate this list.  =)